src/data/setoid/basic.lean

```diff
@@ -36,8 +36,12 @@ variables {α : Type*} {β : Type*}
 /-- A version of `setoid.r` that takes the equivalence relation as an explicit argument. -/
 def setoid.rel (r : setoid α) : α → α → Prop := @setoid.r _ r
 
+/-- A version of `quotient.eq` compatible with `setoid.rel`, to make rewriting possible. -/
+lemma quotient.eq_rel {r : setoid α} {x y} : ⟦x⟧ = ⟦y⟧ ↔ r.rel x y := quotient.eq
```

Comment:
This version looks odd to me: the quotient square brackets are usually when the setoid argument is a typeclass, but here it's implicit...

Reply:
You're quite right, my mistake:

```suggestion
lemma quotient.eq_rel [setoid α] {x y} : ⟦x⟧ = ⟦y⟧ ↔ r.rel x y := quotient.eq
```

Reply:
This change breaks things downstream

Reply:
I just removed the lemma stated this way, it's weird to use `rel` for the typeclass version anyway.

Reply:
This is a horrible can of worms: [Zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Setoid.20hell/near/243323132)